src/repository/FindTreesOptions.ts

```diff
@@ -0,0 +1,11 @@
+/**
+ * Special options passed to TreeRepository#findTrees
+ */
+export interface FindTreesOptions {
+
+    /**
+     * When loading a tree from a TreeRepository, limits the depth of the descendents loaded
+     */
+    depth: number;
```

Comment:
make `depth` optional.

when `depth` is undefined then treat that as the unlimited case.

I think that would be a clearer API & would match other cases where we have optional options

Reply:
Thanks for the suggestion. I agree that it does indeed make more sense.

I have implemented the suggested change, and extended the test to include examples of using empty objects for the new arguments. Let me know if there's anything else you think can be improved here.

Thanks!

Reply:
@imnotjames I just rebased my branch to include the latest changes done to master. Assuming the tests (currently running) pass, is there anything else I can to do help move this PR forward?

Reply:
@imnotjames I've once again rebased my branch, and the tests are passing. Let me know if this PR is relevant to TypeORM, or if not, I can close it. thanks